src/main/java/com/xiaomi/infra/pegasus/rpc/async/ClusterManager.java

```diff
@@ -43,32 +43,25 @@
     logger.info("operating system name: {}", osName);
   }
 
-  public ClusterManager(
-      int timeout,
-      int io_threads,
-      boolean enableCounter,
-      String perfCounterTags,
-      int pushIntervalSecs,
-      String[] address_list)
-      throws IllegalArgumentException {
-    setTimeout(timeout);
-    this.enableCounter = enableCounter;
+  public ClusterManager(ClusterOptions opts) throws IllegalArgumentException {
+    setTimeout(opts.operationTimeout());
+    this.enableCounter = opts.enablePerfCounter();
```

Comment:
I think there is no need to saving too many copies for the member of options. We can simply remain a reference to options here.

Reply:
Only `operationTimeout`, `metaList`, and `enableCounter` are copied as private members of `ClusterManager`, not "too many". I don't wanna change much. This is not my concern in this PR.